src/main/java/com/google/gcp_variant_transforms/library/VariantToBqUtilsImpl.java

```diff
@@ -115,37 +118,42 @@ public void addInfo(TableRow row, VariantContext variantContext,
     for (Genotype genotype : genotypes) {
       TableRow call = new TableRow();
       call.set(Constants.ColumnKeyNames.CALLS_SAMPLE_NAME, genotype.getSampleName());
-      addFormatAndPhaseSet(call, genotype, vcfHeader);
+      addFormat(call, genotype, vcfHeader);
       addGenotypes(call, genotype.getAlleles(), variantContext);
       callRows.add(call);
     }
     return callRows;
   }
 
   public Object convertToDefinedType(Object value, VCFHeaderLineType type, int count) {
-    if (!(value instanceof List)) {
-      // Deal with single value.
+    if (!(value instanceof List) && count != Constants.DEFAULT_REPEATED_FIELD_COUNT) {
```

Comment:
Let's see if we can make the logic of this function more straight forward. How about this:
```
public Object convertToDefinedType(Object value, VCFHeaderLineType type, int count) {
  if ((value instanceof String) && ((String)value).contains(",")) { exactly what you have now ... }

  if (value instanceof List) {
    if (count != Constants.DEFAULT_REPEATED_FIELD_COUNT && count != valueList.size()) { raise }
    iterate over elements of list and convert them
  } else {
    if (count > 1) { raise }
    return convertSingleObjectToDefinedType(value, type); 
  }
```

What do you think?

Reply:
I agree. Parsing first and then deal with list and single value is the better solution.

Reply:
One thing is that if it is a single value with `DEFAULT_REPEATED_COUNT`, we need to handle it as list. I'll change the 
`if (value instanceof List)` to `if ((value instanceof List) || count == DEFAULT_REPEATED_COUNT)`

Reply:
You are right I missed that case. I still like to keep that if statement simple so it's easy to follow the logic. How about this:
```
public Object convertToDefinedType(Object value, VCFHeaderLineType type, int count) {
  if ((value instanceof String) && ((String)value).contains(",")) { exactly what you have now ... }

  if (value instanceof List) {
    if (count != Constants.DEFAULT_REPEATED_FIELD_COUNT && count != valueList.size()) { raise }
    iterate over elements of list and convert them
  } else {
    if (count > 1) { raise }
    convertedValue = convertSingleObjectToDefinedType(value, type); 
    if (count == 1)
      return convertedValue
    else  // DEFAULT_REPEATED_COUNT
      return Arrays.asList(convertedValue)
  }
```

Reply:
If the single value is `"."` with `DEFAULT_REPEATED_COUNT`, we still need to return an empty list instead of `return Arrays.asList(convertedValue)` directly. I think for the `DEFAULT_REPEATED_COUNT` cases, we need to do the same action as list, iterate all elements to add those value `not null` to the result list.